sktime/transformations/series/impute.py

```diff
@@ -70,49 +70,70 @@ def transform(self, Z, X=None):
 
         Parameters
         ----------
-        Z : pd.Series
+        Z : pd.Series, pd.DataFrame
 
         Returns
         -------
-        z : pd.Series
-            Transformed time series.
+        Z : pd.Series, pd.DataFrame
+            Transformed time series(es).
         """
         self.check_is_fitted()
         self._check_method()
-        z = check_series(Z, enforce_univariate=True)
+        Z = check_series(Z)
+        # multivariate
+        if isinstance(Z, pd.DataFrame):
+            for col in Z:
```

Comment:
Just a quick comment: looping over columns doesn't seem like the right approach, it should be possible to make use of more efficient methods with `pd.DataFrames` or the underlying `np.ndarray`s, e.g. the `.fillna(...)` method seems to work on data frames too.

Reply:
yes this is true, just added it whenever applicable.